docs/dataservice-cache-example.md

```diff
@@ -0,0 +1,195 @@
+# Cache Example
+
+On this page, find instructions on how to build and run the example project and get partition data, use mutable and protected cache using the HERE Open Location Platform (OLP) SDK for C++.
+
+Before you run the example project, authorize to HERE OLP:
+
+1. On the [Apps & keys](https://platform.here.com/admin/apps) page, copy your application access key ID and access key secret.
+   For instructions on how to get the access key ID and access key secret, see the [Get Credentials](https://developer.here.com/olp/documentation/access-control/user-guide/topics/get-credentials.html) section in the Terms and Permissions User Guide.
+
+2. In `examples/main.cpp`, replace the placeholders with your access key ID, access key secret, or you can specify values using command line options.
+
+   ```cpp
+   AccessKey access_key{};  // You can put your here.access.key.id and here.access.key.secret
+   ```
+
+## Build and Run on Linux
+
+To build and run the example project on Linux:
+
+1. Enable examples of the `CMake` targets.
+
+   ```bash
+   mkdir build && cd build
+   cmake -DOLP_SDK_BUILD_EXAMPLES=ON ..
+   ```
+
+2. In the **build** folder, build the example project.
+
+   ```bash
+   cmake --build . --target dataservice-example
+   ```
+
+3. Execute the example project.
+
+   ```bash
+   ./examples/dataservice-example --example cache --key_id "here.access.key.id" --key_secret "here.access.key.secret"
+   ```
+
+4. To show help. 
+   ```bash
+   ./examples/dataservice-example --help
+   ```
+
+After building and running the example project, you see the following information:
+
+```bash
+[INFO] protected-cache-example - Mutable cache path is "none"
+[INFO] protected-cache-example - Protected cache path is "/tmp/cata.log_client_example/cache"
+[INFO] ThreadPoolTaskScheduler - Starting thread 'OLPSDKPOOL_0'
+[INFO] CatalogRepository - cache catalog '@0^0' found!
+[INFO] PartitionsCacheRepository - Get 'hrn:here:data::olp-here-test:edge-example-catalog::versioned-world-layer::1::0::partition'
+[INFO] PartitionsRepository - cache data 'versioned-world-layer[1]@0^0' found!
+[INFO] DataRepository - cache data 'versioned-world-layer[1]@0^0' found!
+[INFO] protected-cache-example - Request partition data - Success, data size - 3375
```

Comment:
Just a hint: Using GetData() with FetchOptions::CacheOnly should also do the trick and it would be more obvious to the user.